ls1intum/Artemis
# `Modeling exercises`: Replace modeling participation auto save alert with permanent save information

### Checklist
#### General


#### Client

### Motivation and Context



In modeling exercises, Artemis attempts to auto-save the current state of the model every 30 seconds if something changed. If this succeeds, Artemis is displaying an alert above the modeling editor, causing the editor to be moved down and moved up (if the alert disappears) over and over while someone works on their modeling submission.

I know from practice that this is extremely annoying and interrupts the user workflow as the whole editor is moved 4 times a minute while working on a model; if you're dragging an item while it happens, you miss the target, etc.

See the video below.

### Description


This PR removes the auto-save alert that causes the issue, and instead introduces a permanent state indicator about wether or not all changes are saved. It provides the info whenever the user needs to look it up as opposed to the prior alert that only shows what's happening when it happens. I am keeping all other alerts (e.g. for submission confirmation), but the automated alert 2 times a minute needs to vanish... :)

### Steps for Testing

Prerequisites:
- 1 Student
- 1 Active Modeling Exercise

1. Open the modeling exercise as a student
2. The indicator should be green and say "All changes are saved"
3. Change something. The indicator should change to a yellow "Unsaved changes".
4. Wait for it to save. While saving, it should show "Saving ...". You can look at this by throttling your browser connection speed in dev tools
5. After save, it should again show "All changes are saved".

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage





modeling-editor.component.ts: 49.12% | 53.24%
modeling-submission.component.ts: 74% | 84.83%

### Screenshots



Before (annoying):

https://user-images.githubusercontent.com/23171488/143057471-002b83ed-af32-4ce5-8d16-d8d1a6310d44.mov

After:

https://user-images.githubusercontent.com/23171488/143057549-a5e1b611-8c73-4c2c-a44e-40fa248e3a39.mov
